clearwater-config-manager.root/usr/share/clearwater/clearwater-config-manager/scripts/validate_json_config.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) Metaswitch Networks 2017
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+import jsonschema
+import json
+import sys
+import yaml
+
+# Parse the command line options
+if len(sys.argv) != 3:
+    print "Usage: validate_json_config <schema to validate against> <file to validate>"
+    sys.exit(1)
+
+schema_file = sys.argv[1]
+config_file = sys.argv[2]
+
+# Load the schema and config files
+try:
+    schema = json.load(open(schema_file))
+except IOError:
+    print "Unable to open {}".format(schema_file)
+    sys.exit(1)
+except ValueError as e:
+    print "{} doesn't contain valid JSON ({})".format(schema_file, e.message)
+    sys.exit(1)
+
+try:
+    config = json.load(open(config_file))
+except IOError:
+    print "Unable to open {}".format(config_file)
+    sys.exit(1)
+except ValueError as e:
+    print "{} doesn't contain value JSON ({})".format(config_file, e.message)
+    sys.exit(1)
+
+# Validate the configuration file against the schema
+validator = jsonschema.Draft4Validator(schema)
+error_list=sorted(validator.iter_errors(config), key=lambda e: e.path)
+
+# If there are any errors, we want to print them out in a user friendly fashion,
+# then exit with a failure code.
+
+# For each error, we have a deque that tells us where in the file the error is,
+# and a string that holds the actual error message, e.g.:
+#   "deque([u'hostnames', 0, u'records', u'target'])"
+#   "1 is not of type u'string'"
+#
+# We want to turn this into a list that maps the structure of the JSON file,
+# e.g.:
+#
+# hostnames:
+#   element 1:
+#     name:
+#       errors:
+#       - 1 is not of type 'string'
+#     records:
+#       rrtype:
+#         errors:
+#         - 'CNAME2' does not match '^CNAME$'
+#       target:
+#         errors:
+#         - 1 is not of type 'string'
+#
+# We construct a nested dictionary of the errors, then print it out in a YAML
+# format.
+if error_list:
+    print "{} is not valid.".format(config_file)
+    print "The errors, and the location of the errors in the configuration file, are displayed below:\n"
+
+    temp_dict = {}
+    for error in error_list:
+        nest = temp_dict
+        if len(error.path) == 0:
+            last = "Top level"
+        else:
+            last = error.path.pop()
```

Comment:
I looked at this for like 15 mins and couldn't figure out why you needed to pop the last element off the list. Please could you add a comment?